src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -0,0 +1,26 @@
+package io.iohk.ethereum.db.dataSource
+
+trait DataSource {
+
+  type Key = IndexedSeq[Byte]
+  type Value = IndexedSeq[Byte]
+
+  /**
+    * This function obtains the associated value to a key, if there exists one.
+    *
+    * @param namespace which will be searched for the key.
+    * @param key
+    * @return the value associated with the passed key.
+    */
+  def get(namespace: Byte, key: Key): Option[Value]
+
+  /**
+    * This function updates the DataSource by deleting and inserting new (key-value) pairs.
+    *
+    * @param namespace from which the (key-value) pairs will be removed and inserted.
+    * @param toRemove which includes all the keys to be removed from the DataSource.
+    * @param toUpdate which includes all the (key-value) pairs to be inserted into the DataSource.
```

Comment:
I'm not sure if that would also be a proper name as the keys from `toUpdate` could already be on the `DataSource`, resulting in the update of its value and not insertion. I will improve the description on that function to include that, maybe we could have a name like `toInsertOrUpdate` that will be more general?